### PR TITLE
Improve structure of likelihoods subdirectory

### PR DESCRIPTION
**Pull request by st--**
Following on from #1405.

* New structure underneath `gpflow/likelihoods/`:
  * `base.py`: all base classes (Likelihood, MonteCarloLikelihood, ScalarLikelihood) and SwitchedLikelihood
  * `multiclass.py`: multi-class classification (Softmax, MultiClass + RobustMax)
  * `scalar_continuous.py`: continuous-Y subclasses of ScalarLikelihood (Gaussian, StudentT, Exponential, Beta, Gamma)
  * `scalar_discrete.py`: discrete-Y subclasses of ScalarLikelihood (Bernoulli, Poisson, Ordinal)
  * `utils.py`: the `inv_probit` link function used by Bernoulli and Beta likelihoods
  * `misc.py`: GaussianMC - used for demonstration/tests only.
* Tests for multi-class classification likelihoods moved out into their own test module (including stubs for the missing MultiClass quadrature tests of #1091)
* Re-activates the quadrature tests for ScalarLikelihood subclasses with analytic variational_expectations/predict_log_density/predict_mean_and_var that inadvertently got disabled by #1334 
* Fixes a bug in Bernoulli._predict_log_density that was uncovered by these tests

### CODE REVIEW

File: tests/gpflow/optimizers/test_natural_gradient.py [hunk 1]
```diff
@@ -59,7 +63,7 @@ def sgpr_and_svgp(data, inducing_variable, kernel, likelihood):
     return sgpr, svgp
 
 
-def assert_different(value1, value2, rtol=0.1):
+def assert_different(value1, value2, rtol=0.07):
```
Comment (jameshensman) [inline tests/gpflow/optimizers/test_natural_gradient.py:66]: what causes the change in rtol?

File: tests/gpflow/optimizers/test_natural_gradient.py [hunk 2]
```diff
@@ -12,19 +12,23 @@
 class Setup:
```
Comment (jameshensman) [inline tests/gpflow/optimizers/test_natural_gradient.py:12]: This is a bit out of place in the refactor / reorganization, but in the spirit of "tidy as you go", I think it stays.

**Comment by jameshensman** [comment]
Yikes, what was the bug in predict_log_density? That could have affected a lot of things...

**Comment by st--** [comment]
> Yikes, what was the bug in predict_log_density? That could have affected a lot of things...

#1334 changed the behaviour of predict_log_density to sum over outputs (instead of one number per output per data row), and the Bernoulli likelihood has an analytic _predict_log_density, but hadn't been summing over it...

File: tests/gpflow/optimizers/test_natural_gradient.py (hunk 1 above)
Comment (st--) [reply tests/gpflow/optimizers/test_natural_gradient.py:66]: It's a stochastic test; reshuffling the tests changed the random number generator, and I gave up on trying to find a seed that makes all of them pass...

File: tests/gpflow/optimizers/test_natural_gradient.py (hunk 2 above)
Comment (st--) [reply tests/gpflow/optimizers/test_natural_gradient.py:12]: yeah, part of this was not wanting to have to think too much about how to consistently get tensorflow random numbers - I know there's tf.random.set_seed() but I trusted a RandomState object more...
